### PR TITLE
Implement Filter events by location & category, RSVP to events and reset password

### PR DESCRIPTION
**Pull request by meshnesh**
# What does this PR do?

implement RSVP to an event, reset password and filter events

## Description of tasks to be completed?

* Create rsvp association table
* write tests for user rsvp to an event
* Implement RSVP functions in an event
* Write tests for Reser password
* Validate email and generate token to reset password
* Decode and validate token
* Reset password and direct user to login
* Login with new password
* write tests for possible event filter scenarios
* Write functions for event filter functions

## How should this be tested manually?

*  run the tests: `python manage.py test`

* Then to test the endpoints with postman, start the server and hit the following endpoints:

* First create events, then test filter: 
```
/api/events/all?title=title&location=location&cartegory=category [GET] filter event by specific location, title and category
/api/events/all?title=title [GET] Filter events by title
/api/events/all?cartegory=category [GET] filter events by category
/api/events/all?location=location [GET] filter events by location
```

* Create an account:
`/api/auth/register`

* Reset password, validate the email: 
`/api/auth/reset`

* Pass the token in the header to reset the password:
`/api/auth/reset-password`

* Reserve a seat to an event:
* Login:
`/api/auth/login`
* pass the token in the header, rsvp to a particular event with the event id:
`/api/events/1/rsvp/`

### CODE REVIEW

**Review by betsybeth — COMMENTED** [review]
Good implementation of TDD.

**Comment by betsybeth** [comment]
Practise to write good commit messages

**Review by frankip — COMMENTED** [review]
I like the way you have approached the filter and pagination

File: app/models.py [hunk 1]
```diff
@@ -18,6 +24,8 @@ class User(db.Model):
     password = db.Column(db.String(256))
     eventlists = db.relationship(
         'Events', order_by='Events.id', cascade="all, delete-orphan")
+    myrsvps = db.relationship('Events', secondary=rsvps,
```
Comment (frankip) [inline app/models.py:28]: great association table implementation

**Pull request merged**